### PR TITLE
Making buffers untyped

### PR DESCRIPTION
**Pull request by oierlauzi**
Decided that buffer objects should not have a numerical type associated to them. This should be handled elsewhere. Now, buffers simply represent a contiguous area of memory with the provided size and alignment requirements. In this way, buffers are more flexible in the contents they can store (not limited to the `numerical_type` enum). 

This poses significant changes in the interfaces, everything related to buffers has been changed slightly.

### CODE REVIEW

**Review by MartinSalinas98 — CHANGES_REQUESTED** [review]

File: include/xmipp4/core/compute/host_buffer_visitor.hpp
[the remark this answers is not in the document]
Comment (oierlauzi) [reply include/xmipp4/core/compute/host_buffer_visitor.hpp:1]: As buffers do not have type information now, they can not be casted to a span (it requires type information). In the future, we will need to re-implement this functions with whatever type handling mechanism we decide.

File: src/compute/host/host_transfer.cpp [hunk 1]
```diff
@@ -64,6 +64,7 @@ void host_transfer::transfer_copy(const host_buffer &src_buffer,
 std::shared_ptr<device_buffer> 
 host_transfer::transfer(const std::shared_ptr<host_buffer> &buffer, 
                         device_memory_allocator&,
+                        std::size_t,
```
[the remark this answers is not in the document]
Comment (oierlauzi) [reply src/compute/host/host_transfer.cpp:67]: It is the alignment parameter (not size). As the buffer is casted, it is ignored.

File: tests/unitary/src/compute/test_copy_region.cpp [hunk 2]
```diff
@@ -47,12 +47,3 @@ TEST_CASE( "default construct copy region", "[copy_region]" )
...
 }
-
-TEST_CASE( "convert region to bytes", "[copy_region]" )
-{
-    const copy_region region(4, 6, 10);
-    const auto region_bytes = as_bytes(region, 8U);
-    REQUIRE( region_bytes.get_source_offset() == 32 );
-    REQUIRE( region_bytes.get_destination_offset() == 48 );
-    REQUIRE( region_bytes.get_count() == 80 );
-}
```
[the remark this answers is not in the document]
Comment (oierlauzi) [reply tests/unitary/src/compute/test_copy_region.cpp:58]: Function was deleted. Now regions are expressed in bytes

File: src/memory/aligned_alloc.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace xmipp4
+{
+namespace memory
+{
+
+void* aligned_alloc(std::size_t size, std::size_t alignment) noexcept
+{
+    #if XMIPP4_WINDOWS
+        return _aligned_malloc(size, alignment);
```
[the remark this answers is not in the document]
Comment (oierlauzi) [reply src/memory/aligned_alloc.cpp:43]: Yes: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/aligned-malloc?view=msvc-170#requirements

File: include/xmipp4/core/compute/host_buffer_visitor.hpp
[the remark this answers is not in the document]
Comment (MartinSalinas98) [reply include/xmipp4/core/compute/host_buffer_visitor.hpp:1]: And, for now, we are just "reducing" functionality to be re-added later when we decide about how to implement it?

[the remark this answers is not in the document]
Comment (oierlauzi) [reply include/xmipp4/core/compute/host_buffer_visitor.hpp:1]: Yes

File: src/memory/aligned_alloc.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (oierlauzi) [reply src/memory/aligned_alloc.cpp:43]: Yes: https://learn.microsoft.com/es-es/cpp/c-runtime-library/reference/aligned-malloc?view=msvc-170#requirements

**Review by MartinSalinas98 — APPROVED** [review]

**Pull request merged**